spec/aruba/api/core_spec.rb

```diff
@@ -0,0 +1,138 @@
+require 'spec_helper'
+require 'aruba/api'
+require 'fileutils'
+
+describe Aruba::Api::Core do
+  include_context 'uses aruba API'
+
+  describe '#expand_path' do
+    context 'when file_name is given' do
+      it { expect(@aruba.expand_path(@file_name)).to eq File.expand_path(@file_path) }
+    end
+
+    context 'when file_path is given' do
+      it { expect(@aruba.expand_path(@file_path)).to eq @file_path }
+    end
+
+    context 'when path contains "."' do
+      it { expect(@aruba.expand_path('.')).to eq File.expand_path(aruba.current_directory) }
+    end
+
+    context 'when path contains ".."' do
+      it { expect(@aruba.expand_path('path/..')).to eq File.expand_path(File.join(aruba.current_directory)) }
+    end
+
+    context 'when path is nil' do
+      it { expect { @aruba.expand_path(nil) }.to raise_error ArgumentError }
+    end
+
+    context 'when path is empty' do
+      it { expect { @aruba.expand_path('') }.to raise_error ArgumentError }
+    end
+
+    context 'when dir_path is given similar to File.expand_path ' do
+      it { expect(@aruba.expand_path(@file_name, 'path')).to eq File.expand_path(File.join(aruba.current_directory, 'path', @file_name)) }
+    end
+
+    context 'when file_name contains fixtures "%" string' do
+      let(:runtime) { instance_double('Aruba::Runtime') }
+      let(:config) { double('Aruba::Config') }
+      let(:environment) { instance_double('Aruba::Environment') }
+
+      let(:klass) do
+        Class.new do
+          include Aruba::Api
+
+          attr_reader :aruba
+
+          def initialize(aruba)
+            @aruba = aruba
+          end
+        end
+      end
+
+      before :each do
+        allow(config).to receive(:fixtures_path_prefix).and_return('%')
+        allow(config).to receive(:root_directory).and_return aruba.config.root_directory
+        allow(config).to receive(:working_directory).and_return aruba.config.working_directory
+      end
+
+      before :each do
+        allow(environment).to receive(:clear)
```

Comment:
Even if this returns `nil` I'd put it in. Because otherwise this allows errors to spawn (Not sure on likelihood)

Reply:
I'm think I'm going to rewrite this spec because it contains way too much mocking. I copied it straight from master so I'll change it there also.